backend.py

```diff
@@ -532,6 +550,54 @@ def configureNTP(mounts, ntp_config_method, ntp_servers):
     util.runCmd2(['chroot', mounts['root'], 'systemctl', 'enable', 'chronyd'])
     util.runCmd2(['chroot', mounts['root'], 'systemctl', 'enable', 'chrony-wait'])
 
+# Setup a new SW RAID device using mdadm
+# The primary-disk (/dev/md/*) is built from the physical disks provided in the answerfile
+def setupSWRAIDDevice(disk_label_suffix, physical_disks, guest_disks):
+    primary_disk = "/dev/md/xs-" + disk_label_suffix
+
+    # Stop the multi-device if it exists
+    if os.path.exists(primary_disk):
+        diskutil.stopSWRAID(primary_disk)
+
+    # Stop any multi-devices using the physical disks we require
+    for device in getMdDevicesUsing(physical_disks):
+        diskutil.stopSWRAID(device)
+
+    # Zero any superblocks on the physical disks
+    for disk in physical_disks:
+        util.runCmd2(['mdadm', '--zero-superblock', disk])
+
+    # Limit SWRAID size to 95%
+    rc, out = util.runCmd2(['blockdev', '--getsize64', physical_disks[0]], with_stdout=True)
+    if rc != 0:
+        raise RuntimeError("Failed to determine size of block device: '%s'" % physical_disks[0])
+
+    sizeGB = int(out.strip()) / (1024 ** 3)
+    swraidSizeGB = int(sizeGB * 0.95)
+
+    # Create multi-device with the first physical disk
+    rc = util.runCmd2(['mdadm', '--create', primary_disk, '--size=%sG' % swraidSizeGB, '--metadata=1.0', '--level=mirror', '--raid-devices=2', '--run', physical_disks[0], 'missing'])
+    if rc != 0:
+        raise RuntimeError("Failed to create SWRAID device: '%s' from initial device: '%s'" % (primary_disk, physical_disks[0]))
+
+    # Add second disk to SW RAID device
+    rc = util.runCmd2(['mdadm', '--manage', primary_disk, '--add', physical_disks[1], '--run'])
+    if rc != 0:
+        raise RuntimeError("Failed to add second disk: '%s' to SWRAID device: '%s'" % (physical_disks[1], primary_disk))
+
+    with open('/proc/sys/dev/raid/speed_limit_max', 'w') as speed_file:
+        speed_file.write(str(constants.swraid_speed_write_max))
+
+    primary_disk = os.path.realpath(primary_disk)
+
+    # Update guest-disks
+    if any(disk in physical_disks for disk in guest_disks):
+        guest_disks = list(filter(lambda disk: disk not in physical_disks, guest_disks))
```

Comment:
I think a list comprehension is preferred in these scenarios. E.g. `guest_disks = [disk for disk in guest_disks if disk not in physical_disks]`